### PR TITLE
Update BindableLayout.cs to fix #5213

### LINKED ISSUE #5213: BindableLayout throws IndexOutOfRange exception during Remove action from ObservableCollection
### Description

I have a StackLayout using the BindableLayout to create a repeater-like control with its ItemSource set to an ObservableCollection. When the items get removed from the collection an IndexOutOfRange exception is thrown. I believe the collection should be sending a reset action instead of remove but as the collection is from realm.io I don't have any control of the action it performs.

### Steps to Reproduce

1. Bind StackLayout BindableLayout to ObservableCollection
2. ObservableCollection triggers CollectionChanged with Action = Remove, OldStartingIndex = 0, and OldItems = All Items
3. IndexOutOfRange exception caused by code at https://github.com/xamarin/Xamarin.Forms/blob/15c6f561e6d51071b7b3cdd9e92d2cbf16edaf2b/Xamarin.Forms.Core/BindableLayout.cs#L228

### Expected Behavior

All children are removed from StackLayout.

### Actual Behavior

IndexOutOfRange exception is thrown

### Basic Information

- Version with issue: 3.5.0.129452
- Last known good version: N/A
- IDE: Visual Studio 2017 Professional
- Platform Target Frameworks: 
  - iOS:  12.1
  - Android: 8.1
- Nuget Packages: Xamarin.Forms
- Affected Devices: All

### PR DESCRIPTION
**Pull request by jsmarcus**
### Description of Change ###

Fixed the ItemSourceCollectionChanged method to not advance the index when removing children during a Remove action.

### Issues Resolved ### 

- fixes #5213 

### API Changes ###
 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Tested on Android/iOS.
1. StackLayout with BindableLayout ItemSource set to ObservableCollection.
2. Add 2+ items to the collection.
3. Remove all items from the collection (where the NotifyCollectionChangedAction is Remove, the OldStartingIndex = 0, and the OldItems are all the items in the collection).

### PR Checklist ###

- [ ] Has automated tests 
- [ ] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

**Comment by samhouts** [comment]
@jsmarcus Can you rebase to the 3.5.0 branch? Thank you!

File: Xamarin.Forms.Core/BindableLayout.cs [hunk 1]
```diff
@@ -223,9 +223,9 @@ void ItemsSourceCollectionChanged(object sender, NotifyCollectionChangedEventArg
 					{
 						if (e.OldStartingIndex == -1)
 							goto case NotifyCollectionChangedAction.Reset;
-						for (int i = 0; i < e.OldItems.Count; ++i)
+						for (int i = 0; i < e.OldItems.Count; i++)
 						{
```
Comment (andreinitescu) [inline Xamarin.Forms.Core/BindableLayout.cs]: I wanted to fix it today, but thanks for your help.
I realized my mistake and a better fix I think is to actually remove the `for` iteration and replace with a `while`, using the index and increment it is not necessary:
`while(e.OldItems.Count > 0)
     layout.Children.RemoveAt(0);

Comment (andreinitescu) [reply Xamarin.Forms.Core/BindableLayout.cs]: Also, the unit test must be added

**Comment by andreinitescu** [comment]
Unit test should be added

**Comment by rmarinho** [comment]
Please add a unit test and retarget branch

**Comment by jsmarcus** [comment]
I think I screwed this up. Let me do a new one.

**Comment by jsmarcus** [comment]
Redid this PR now at #5243

Added a unit test and rebased to 3.5.0

**Pull request closed without merging**